Mage/src/main/java/mage/ConditionalMana.java

```diff
@@ -220,4 +227,42 @@ public void add(ManaType manaType, int amount) {
                 break;
         }
     }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(super.hashCode(), conditions, staticText, scope, manaProducerId, manaProducerOriginalId);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        // Check Mana.equals(). If that's isn't equal no need to check further.
+        if (!super.equals(o)) {
+            return false;
+        }
+        ConditionalMana that = (ConditionalMana) o;
```

Comment:
The variable names `o` and `that` don't do much to clarify what these variables are used for imo.

Reply:
Changed `o` to `obj`.
`that` is used as a natural sounding comparison between two values `this` vs `that`.

Reply:
Personally I'd name them something like `obj` -> `objectToCheck` and `that` -> `conditionalManaToCheck` but that's just my opinion. Curious what others think here.